src/components/Dashboard/FirmsAddressList.vue

```diff
@@ -34,14 +34,14 @@
         <div v-else class="pt-4 pb-4">
           <v-list class="pt-0 pb-0" v-if="businessAddress">
             <!-- Delivery Address -->
-            <v-list-item class="delivery-address-list-item" v-if="businessAddress.deliveryAddress">
+            <v-list-item class="delivery-address-list-item">
               <v-list-item-icon class="address-icon mr-0">
                 <v-icon color="primary">mdi-truck</v-icon>
               </v-list-item-icon>
               <v-list-item-content>
                 <v-list-item-title class="mb-2 address-title">Delivery Address</v-list-item-title>
                 <v-list-item-subtitle>
-                  <ul class="address-subtitle pre-line">
+                  <ul class="address-subtitle pre-line" v-if="businessAddress.deliveryAddress">
```

Comment:
Did you test with mock data? What does the data look like when there is no business address?
- does the app load or throw an error while fetching data?
- does the conditional on line 35 pass?

Reply:
- When there is no business address, then It displays "Complete your filing to display" for all components.
- When there is data but delivery address or mailing address is null, then it displays "Not Entered".

I had the same question too. In what scenario do we ever run into "Not Entered" for delivery/mailing address, if create-ui is error-proof already.

Reply:
I did test it with mock data and I am writing a unit tests for those components. I always forget to write unit test.

Reply:
1. We may not have the info if the firm was loaded directly from the old system. That's why we need to do conversion filings.

2. Your test data assumes that there is a business object that contains a null delivery address and a null mailing address. Is that actually what is in the store when one of these incomplete firms is loaded in Filings UI?

Reply:
Looking at Argus's file, I think `delivery_address` or `mailing_address` can be `null`. From that file, There are several cases where `delivery_address` is null but `mailing_address` is not null.

Reply:
Could `businessAddress` be null?

Reply:
`businessAddress` can be null, but it is handled on layer up in line 35.

Reply:
I see the conditional on line 35, but I don't see the "Not Entered" message in this situation.

Reply:
I added the v-else for the case where businessAddress is null in my latest commit.

Reply:
OK, this is very safe code now :)

Also, great job on the unit tests where you verify all possible scenarios 👍